google-ads-migration-examples/src/main/java/com.google.ads.googleads.migration/createcampaign/CreateCompleteCampaignAdWordsApiOnly.java

```diff
@@ -0,0 +1,429 @@
+
+    campaign.setAdvertisingChannelType(AdvertisingChannelType.SEARCH);
+
+    // Set the campaign network options to Search and Search Network.
+    NetworkSetting networkSetting = new NetworkSetting();
+    networkSetting.setTargetGoogleSearch(true);
+    networkSetting.setTargetSearchNetwork(true);
+    networkSetting.setTargetContentNetwork(false);
+    networkSetting.setTargetPartnerSearchNetwork(false);
+    campaign.setNetworkSetting(networkSetting);
+
+    // Create the operation.
+    CampaignOperation operation = new CampaignOperation();
+    operation.setOperand(campaign);
+    operation.setOperator(Operator.ADD);
+
+    CampaignOperation[] operations = new CampaignOperation[] {operation};
+
+    // Add the campaign.
+    CampaignReturnValue result = campaignService.mutate(operations);
+
+    Campaign campaignResult = result.getValue(0);
+    // Display the campaign.
+    System.out.printf("Campaign with ID '%d' and name %s was created.%n",
+      campaignResult.getId(),
+      campaignResult.getName());
+    return campaignResult;
+  }
+
+  /**
+   * Creates a budget.
+   *
+   * @param adWordsServices the Google AdWords services interface.
+   * @param session the client session.
+   * @throws RemoteException if the API request failed due to other errors.
+   */
+  private Budget createBudget(AdWordsServicesInterface adWordsServices,
+                                      AdWordsSession session) throws RemoteException {
+    // Get the BudgetService.
+    BudgetServiceInterface budgetService =
+      adWordsServices.get(session, BudgetServiceInterface.class);
+
+    // Create a budget, which can be shared by multiple campaigns.
+    Budget sharedBudget = new Budget();
+    sharedBudget.setName("Interplanetary Cruise #" + System.currentTimeMillis());
+    Money budgetAmount = new Money();
+    budgetAmount.setMicroAmount(500_000L);
+    sharedBudget.setAmount(budgetAmount);
+    sharedBudget.setDeliveryMethod(BudgetBudgetDeliveryMethod.STANDARD);
+
+    BudgetOperation budgetOperation = new BudgetOperation();
+    budgetOperation.setOperand(sharedBudget);
+    budgetOperation.setOperator(Operator.ADD);
+
+    BudgetOperation[] operations = new BudgetOperation[] {budgetOperation};
+
+    // Add the budget.
+    BudgetReturnValue result = budgetService.mutate(operations);
+    Budget budgetResult = result.getValue(0);
+    // Display budget.
+    System.out.printf("Budget with ID %d and name %s was created.%n",
+      budgetResult.getBudgetId(),
+      budgetResult.getName());
+    return budgetResult;
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param adWordsServices the Google AdWords services interface.
+   * @param session the client session.
+   * @throws RemoteException if the API request failed due to other errors.
+   * @throws UnsupportedEncodingException if encoding the final URL failed.
+   */
+  private void runExample(AdWordsServicesInterface adWordsServices, AdWordsSession session)
```

Comment:
It would help readability if `runExample` was just below `main`. Then readers can scroll from the top and quickly see the overall approach before reading the method for each step.

Reply:
@jradcliff that makes sense to me, but I should point out that all of our other examples follow the pattern used here. Happy to change it- let me know what you think.

Reply:
If we have other ones that put `runExample` at the end, could you file an issue to have those fixed? It hinders readability when the flow of high level methods requires jumping around in the source file.

Reply:
Done (will those report issues in the future).